fix: clarify usage of dialog value for the method attribute (#10569)
mdn/content: files/en-us/web/html/element/form/index.md

    - `post`: The [POST method](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.5); form data sent as the [request body](/en-US/docs/Web/API/Request/body).
    - `get`: The [GET method](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.3); form data appended to the `action` URL with a `?` separator. Use this method when the form [has no side-effects](/en-US/docs/Glossary/Idempotent).

>     - `dialog`: When the form is inside a {{HTMLElement("dialog")}}, closes the dialog on submission. This is a means to close the `dialog` element without the need for JavaScript. Note that no form data is submitted so, specifying an `action` attribute will have no effect.

Author: The sentence is a little wonky. Basically what I want to get across is this piece from the specification:

"submitting the form is intended to close the dialog box in which the form finds itself, if any, and otherwise not submit."

>     - `dialog`: When the form is inside a {{HTMLElement("dialog")}}, closes the dialog on submission. This is a means to close the `dialog` element without the need for JavaScript. Note that no form data is submitted so, specifying an `action` attribute will have no effect.

Reviewer: ```suggestion
    - `dialog`: When the form is inside a {{HTMLElement("dialog")}}, closes the dialog on and throws a submit event on submission without submitting data or clearing the form
```

Reviewer: i moved the action part to the action section (see my commit)
